packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -90,27 +168,30 @@ class CupertinoTabScaffold extends StatefulWidget {
   /// Creates a layout for applications with a tab bar at the bottom.
   ///
   /// The [tabBar] and [tabBuilder] arguments must not be null.
-  const CupertinoTabScaffold({
+  CupertinoTabScaffold({
     Key key,
     @required this.tabBar,
     @required this.tabBuilder,
+    this.controller,
     this.backgroundColor,
     this.resizeToAvoidBottomInset = true,
   }) : assert(tabBar != null),
        assert(tabBuilder != null),
+       assert(controller == null || controller.index < tabBar.items.length),
```

Comment:
Keep them as separate asserts. Makes them more readable when the user hits them (since this assert shows up in their error message).

Reply:
These 2 are combined by an `or`, not sure how to separate them into 2 assertions

Reply:
I just meant writing them as 2 separate asserts

When users run into it, it'll show up as something like:

```
The following assert failed:

assert(controller == null);
```

Then it's super clear what the user needs to do.